### ISSUE TITLE
PushSecret: option to add additional tags\labels

### ISSUE THREAD

**Issue by tabnul** [external-secrets/external-secrets#1821]
**Describe the solution you'd like**
I would like to be able to manage the tags that are set on the external secret when using PushSecret.

**What is the added value?**
Tags can be used for governance, compliance, and billing purposes and in a lot of organizations it is mandatory to add certain tags to resources.
Currently, only the 'managed-by: external-secrets' is set to (AWS Secrets Manager in our case).

**Comment by dvbthijsvnuland** [comment]
For us , this functionality is a blocker to actually start using the push secret functionality.
In fact, i know from other companies in the past, that tagging AWS resources is mandatory.

**Comment by moolen** [comment]
Agree, this is widely used. Tho to make that work we need to support multiple "dimensions". E.g. in GCP Secret Manager we can define both `annotations` and `labels`. The same is true for the Kubernetes provider.

I'd be happy to review a PR if someone takes this on :pray:.

**Comment by shuheiktgw** [comment]
I've started thinking about this feature. Let me create a PR to discuss the possible design!

**Comment by shuheiktgw** [comment]
Hi, I've created a PR to discuss the potential API design! Any feedback will be much appreciated 🙂 
https://github.com/external-secrets/external-secrets/pull/2600

**Comment by DarrenBishop** [comment]
@shuheiktgw perhaps this should also be a SecretsStore matter i.e. where provider specific concerns are usually addressed?

Maybe a _'template'_ and _'templateParameters'_ subsections, for SecretStore and PushSecret, respectively?

To keep PushSecret decoupled from the SecretStore-provider, the SecretStore-template could feature placeholders in-line e.g. in annotations and/or labels; the PushSecret-templateParameters then just gives values for those placeholders.

**Comment by ivankatliarchuk** [comment]
Not fully resolved. Need to support update as well
